### PR TITLE
Unify logging in tests

### PR DESCRIPTION
**Pull request by adambabik**
I noticed that our logging strategy in tests, in terms of selected logger and logs format, is not unified. This change fixes this.

There is still one problem when byte slices are not hex-encoded when displayed. It's a situation where the logger is used this way: 
```go
logger := app.logger.WithFields("hash", zap.Binary("messageID", m.ID()))
logger.Debug("some message")
```

In this case, `messageID` field will not be hex-encoded.

This will work fine:
```go
logger.Debug("some message", zap.Binary("messageID", m.ID()))
```

I described the problem in https://github.com/uber-go/zap/issues/730.

### CODE REVIEW

**Comment by PombeirP** [comment]
> There is still one problem when byte slices are not hex-encoded when displayed. It's a situation where the logger is used this way:
> 
> ```go
> logger := app.logger.WithFields("hash", zap.Binary("messageID", m.ID()))
> logger.Debug("some message")
> ```
> 
> In this case, `messageID` field will not be hex-encoded.
> 
> This will work fine:
> 
> ```go
> logger.Debug("some message", zap.Binary("messageID", m.ID()))
> ```

@adambabik Isn't it happening the other way around? I tested this branch, and see this:

```
2019-07-30T10:52:27.342+0200    DEBUG   sharedsecret/sharedsecret.go:53 saving a shared key     {"user": "bob", "SharedSecret": {"site": "generate", "their-public-key": "BOpaFoVJ/nSvfTE48iiqyZdNz5E1PXPWbdS3DrV3HQ3vUQUxoIF0l8zWopHL+a7upLCKSuU8Frm75argTLlKdVc=", "installation-id": "bob1"}}
2019-07-30T10:52:27.342+0200    DEBUG   sharedsecret/sharedsecret.go:99 no shared secret for installation       {"user": "bob", "SharedSecret": {"site": "Agreed", "installation-id": "alice1"}}
2019-07-30T10:52:27.342+0200    DEBUG   encryption/protocol.go:257      shared secret agreement {"user": "bob", "Protocol": {"site": "BuildDirectMessage", "public-key": "0x04ea5a168549fe74af7d3138f228aac9974dcf91353d73d66dd4b70eb5771d0def510531a0817497ccd6a291cbf9aeeea4b08a4ae53c16b9bbe5aae04cb94a7557", "has-shared-secret": true, "agreed": false}}
```

Looking at the code, `SharedSecret.generate` is using:

```go
	logger.Debug(
		"saving a shared key",
		zap.Binary("their-public-key", crypto.FromECDSAPub(theirPublicKey)),
		zap.String("installation-id", installationID),
	)
```

and `BuildDirectMessage` (encoded with 0x) is using:

```go
	logger := p.logger.With(
		zap.String("site", "BuildDirectMessage"),
		zap.Binary("public-key", crypto.FromECDSAPub(publicKey)),
	)
```

File: zaputil/encoder.go [hunk 1]
```diff
@@ -34,3 +39,29 @@ func RegisterJSONHexEncoder() error {
...
+		Encoder: consoleEncoder,
+	}
+}
+
+func (enc *consoleHexEncoder) AddBinary(key string, val []byte) {
+	enc.AddString(key, "0x"+hex.EncodeToString(val))
+}
+
+func (enc *consoleHexEncoder) Clone() zapcore.Encoder {
+	encodeClone := enc.Encoder.Clone()
```
Comment (PombeirP) [inline zaputil/encoder.go]:
```suggestion
	encoderClone := enc.Encoder.Clone()
```

File: zaputil/encoder_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	tmpFile, err := ioutil.TempFile("", "")
+	require.NoError(t, err)
+
+	cfg := zap.NewDevelopmentConfig()
+	cfg.OutputPaths = []string{tmpFile.Name()}
+	cfg.Encoding = "json-hex"
+	l, err := cfg.Build()
+	require.NoError(t, err)
+
+	l.With(zap.Binary("some-filed", []byte{0x01, 0x02, 0x03})).Debug("test message")
```
Comment (PombeirP) [inline zaputil/encoder_test.go]:
```suggestion
	l.With(zap.Binary("some-field", []byte{0x01, 0x02, 0x03})).Debug("test message")
```

File: zaputil/encoder_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	tmpFile, err := ioutil.TempFile("", "")
+	require.NoError(t, err)
+
+	cfg := zap.NewDevelopmentConfig()
+	cfg.OutputPaths = []string{tmpFile.Name()}
+	cfg.Encoding = "console-hex"
+	l, err := cfg.Build()
+	require.NoError(t, err)
+
+	l.With(zap.Binary("some-filed", []byte{0x01, 0x02, 0x03})).Debug("test message")
```
Comment (PombeirP) [inline zaputil/encoder_test.go]:
```suggestion
	l.With(zap.Binary("some-field", []byte{0x01, 0x02, 0x03})).Debug("test message")
```

**Comment by adambabik** [comment]
@PombeirP oh right. I mixed the order of the snippets. Thanks!

**Pull request merged**